### PR TITLE
PLAT-47266 update corretto casks for 8, 11, and 17

### PR DESCRIPTION
**Pull request by HarryWeppner**
Snapshot of [homebrew cask versions](https://github.com/Homebrew/homebrew-cask-versions)
- 8.342.07.3
- 11.0.16.9.1
- 17.0.4.9.1

### CODE REVIEW

**Comment by HarryWeppner** [comment]
@briangerholdc3, FYI, casks are macOS only.

**Comment by HarryWeppner** [comment]
Successfully built 7.* and V8 on macOS (x86).

File: Casks/corretto17.rb [hunk 1]
```diff
@@ -16,9 +17,8 @@
 
   livecheck do
     url "https://corretto.aws/downloads/latest/amazon-corretto-#{version.major}-#{arch}-macos-jdk.pkg"
-    strategy :header_match do |headers|
-      headers["location"][%r{/amazon-corretto-(\d+(?:\.\d+)+)-macosx-#{arch}\.pkg}i, 1]
-    end
+    regex(%r{/amazon-corretto-(\d+(?:\.\d+)+)-macosx-#{arch}\.pkg}i)
+    strategy :header_match
```
Comment (briangerholdc3) [inline Casks/corretto17.rb:21]: I see that the net effect of this change is to remove the regex match "location", but I don't really understand why it was there, or why it is no longer warranted.  Just asking for my own edification.

**Review by briangerholdc3 — APPROVED** [review]
Looks good.

File: Casks/corretto17.rb (hunk 1 above)
Comment (HarryWeppner) [reply Casks/corretto17.rb:21]: @briangerholdc3 this isn't our code. It simply snapshots the cask definitions from the `homebrew-versions` repo.

**Pull request merged**